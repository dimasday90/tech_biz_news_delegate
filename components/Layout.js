import Navbar from "./Navbar";
import Footer from "./Footer";
import Head from "next/head";
import { Container } from "react-bootstrap";

export default function Layout(props) {
  return (
    <div>
      <Head>
        <title>Tech Biz News</title>
        <link
          rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css"
          integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh"
          crossorigin="anonymous"
        />
      </Head>
      <Navbar />
      <Container fluid style={{ marginTop: 100, marginBottom: 30 }}>
        {props.children}
      </Container>
      <Footer />
    </div>
  );
}
