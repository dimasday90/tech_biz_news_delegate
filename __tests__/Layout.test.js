import React from "react";
import { render, cleanup } from "@testing-library/react";
import * as nextRouter from "next/router";
import renderer from "react-test-renderer";

import Layout from "../components/Layout";

afterEach(cleanup);

describe("Footer Test", () => {
  test("should render footer text", () => {
    nextRouter.useRouter = jest.fn();
    nextRouter.useRouter.mockImplementation(() => ({ route: "/" }));
    const { getByText } = render(<Layout />);
    expect(getByText("Tech Biz News")).toBeInTheDocument();
  });
});

describe("With Snapshot Testing", () => {
  it("should match layout coding", () => {
    nextRouter.useRouter = jest.fn();
    nextRouter.useRouter.mockImplementation(() => ({ route: "/" }));
    const component = renderer.create(<Layout />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
