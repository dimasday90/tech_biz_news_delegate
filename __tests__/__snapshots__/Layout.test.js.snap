// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing should match layout coding 1`] = `
<div>
  <nav
    className="justify-content-between navbar navbar-expand-md navbar-dark bg-dark fixed-top"
    style={
      Object {
        "minHeight": 77,
      }
    }
  >
    <span
      className="navbar-brand"
    >
      Tech Biz News
    </span>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <div
          className="nav-item"
        >
          <button
            className="btn btn-outline-light"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "marginRight": 10,
              }
            }
            type="button"
          >
            Home
          </button>
        </div>
        <div
          className="nav-item"
        >
          <button
            className="btn btn-outline-light"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "marginRight": 10,
              }
            }
            type="button"
          >
            Technology
          </button>
        </div>
        <div
          className="nav-item"
        >
          <button
            className="btn btn-outline-light"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "marginRight": 10,
              }
            }
            type="button"
          >
            Business
          </button>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container-fluid"
    style={
      Object {
        "marginBottom": 30,
        "marginTop": 100,
      }
    }
  />
  <nav
    className="justify-content-center navbar navbar-expand navbar-dark bg-dark"
  >
    <span
      className="navbar-brand"
    >
      Powered by
       
      <a
        href="https://newsapi.org"
        target="_blank"
      >
        newsapi.org
      </a>
    </span>
  </nav>
</div>
`;
